runhouse/resources/hardware/on_demand_cluster.py

```diff
@@ -630,14 +519,15 @@ def ssh(self, node: str = None):
             cmd = f"kubectl exec -it {pod_name} -- /bin/bash"
             subprocess.run(cmd, shell=True, check=True)
 
-        elif node is None:
-            # SSH onto head node - can provide the name as specified in the local ~/.ssh/config
-            subprocess.run(["ssh", f"{self.name}"])
         else:
             # If SSHing onto a specific node, which requires the default sky public key for verification
-            sky_key = Path(self.DEFAULT_KEYFILE).expanduser()
+            ssh_user = self.ssh_creds.get("ssh_user")
+            node = self.ips[0] or self.address
```

Comment:
```suggestion
            node = node or self.address
```
address is extracted from IPs - if not specified use the head node IP